### PR TITLE
new: add selectors in hookpolicy to link automations

### PR DESCRIPTION
**Pull request by t00f**
Adding a selectors property which is a tag expression to select 1 automation in case an endpoint is not specified.

### CODE REVIEW

File: custom_validations.go [hunk 1]
```diff
@@ -879,3 +879,17 @@ func ValidateStringListNotEmpty(attribute string, slice []string) error {
 
 	return nil
 }
+
+// ValidateHookPolicy validates a hook policy.
+func ValidateHookPolicy(policy *HookPolicy) error {
+
+	if policy.Endpoint == "" && len(policy.Selectors) == 0 {
```
Comment (primalmotion) [inline custom_validations.go:886]: I would only allow this if it's set to call a web hook or a remote call in addition

Comment (t00f) [reply custom_validations.go:886]: Actually, we can't know here - Webhook and RemoteCall are information about the automation...
Here we are only verifying the hook policy.

Comment (primalmotion) [reply custom_validations.go:886]: Right

File: specs/hookpolicy.spec [hunk 2]
```diff
@@ -169,3 +183,15 @@ attributes:
     stored: true
     validations:
     - $writeoperations
+
+  - name: triggerType
```
Comment (primalmotion) [inline specs/hookpolicy.spec:187]: This is not a trigger

It should be call EndpointType: URL|Automation

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "link_hook_automations",
    "component": "backend",
    "pr-id": "724",
    "commit-sha": "76f4707aa3c36841c68a48714383efc343467ff9",
    "pipeline": "master"
  },
  {
    "project": "link_hook_automations",
    "component": "gaia",
    "pr-id": "468",
    "commit-sha": "0a5a78e14160575dec9af421e6590822b18cba3f",
    "pipeline": "master"
  },
  {
    "project": "link_hook_automations",
    "component": "underwater",
    "pr-id": "103",
    "commit-sha": "b0cff64c0a4e6592852ab381f539830616f5903a",
    "pipeline": "master"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "link_hook_automations",
    "component": "backend",
    "pr-id": "724",
    "commit-sha": "76f4707aa3c36841c68a48714383efc343467ff9",
    "pipeline": "master"
  },
  {
    "project": "link_hook_automations",
    "component": "gaia",
    "pr-id": "468",
    "commit-sha": "ca91b8c1868e5bfaead5495de51b32541b6ea532",
    "pipeline": "master"
  },
  {
    "project": "link_hook_automations",
    "component": "underwater",
    "pr-id": "103",
    "commit-sha": "6853c0f0f4aab06cdca5df7c591812daf6ca547f",
    "pipeline": "master"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "link_hook_automations",
    "component": "backend",
    "pr-id": "724",
    "commit-sha": "3ed6a018ee120d6a2319961bc30d7b6af0c4083e",
    "pipeline": "master"
  },
  {
    "project": "link_hook_automations",
    "component": "gaia",
    "pr-id": "468",
    "commit-sha": "ca91b8c1868e5bfaead5495de51b32541b6ea532",
    "pipeline": "master"
  },
  {
    "project": "link_hook_automations",
    "component": "underwater",
    "pr-id": "103",
    "commit-sha": "8ee8e23e96cd0b7fdfd0056e55c7a0cf7b431615",
    "pipeline": "master"
  }
]
```

File: specs/hookpolicy.spec (hunk 2 above)
Comment (t00f) [reply specs/hookpolicy.spec:187]: rrraaaaah! je viens de voir ce commentaire :(

Comment (t00f) [reply specs/hookpolicy.spec:187]: @primalmotion all unittests and FT have passed - I am doing this change now but will probably not do all the FT again

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "link_hook_automations",
    "component": "backend",
    "pr-id": "724",
    "commit-sha": "7139b394ba3fbafdea76f353fce94cbeeae52495",
    "pipeline": "master"
  },
  {
    "project": "link_hook_automations",
    "component": "gaia",
    "pr-id": "468",
    "commit-sha": "615f539fcaacb7361e77ab866495dec3d01dbaa9",
    "pipeline": "master"
  },
  {
    "project": "link_hook_automations",
    "component": "underwater",
    "pr-id": "103",
    "commit-sha": "98fac6f61815dceacf455d36e840d010fc5617be",
    "pipeline": "master"
  }
]
```

File: specs/hookpolicy.spec (hunk 2 above)
Comment (primalmotion) [reply specs/hookpolicy.spec]: Sure

**Pull request merged**